lightning/src/chain/channelmonitor.rs

```diff
@@ -1215,12 +1211,8 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 	/// (C-not exported) because we have no HashMap bindings
 	pub fn get_outputs_to_watch(&self) -> &HashMap<Txid, Vec<Script>> {
 		#[cfg(any(test, feature = "fuzztarget"))]
-		for (txid, &(_, ref outputs)) in self.counterparty_commitment_txn_on_chain.iter() {
-			let watched_outputs = self.outputs_to_watch.get(txid).expect("Counterparty commitment txn which have been broadcast should have outputs registered");
-			assert_eq!(watched_outputs.len(), outputs.len());
-			for (watched, output) in watched_outputs.iter().zip(outputs.iter()) {
-				assert_eq!(watched, output);
-			}
+		for (txid, _) in self.counterparty_commitment_txn_on_chain.iter() {
+			self.outputs_to_watch.get(txid).expect("Counterparty commitment txn which have been broadcast should have outputs registered");
```

Comment:
Should we expect even in production code ? If we have at least a counterparty commitment tx onchain their outputs should be guaranteed to be tracked, just grep the struct, we always write watch_ouputs around and the former should always map in the latter. Otherwise that would be a logic bug.

Reply:
Right, I'd skipped it purely because adding assertions in prod is more worrisome than not plus its not ideal to do things that take more time in prod, but I think you're right - `counterparty_commitment_txn_on_chain` should always be 1/2 entries, so it should be ~free, and its basically trivially true. I can just remove the cfg.